vault/wrapping.go

```diff
@@ -184,6 +184,11 @@ DONELISTHANDLING:
 		resp.WrapInfo.WrappedAccessor = resp.Auth.Accessor
 	}
 
+	// Store the accessor of the approle secret in WrappedAccessor
+	if secretIdAccessor, ok := resp.Data["secret_id_accessor"]; ok && resp.Auth == nil {
```

Comment:
Another check that we discussed and would be nice to have, is a [logical.Request.MountType](https://github.com/hashicorp/vault/blob/main/sdk/logical/request.go#L150) == "approle" check here, to be sure we don't do this for a different backend that also uses a field called `secret_id_accessor`. wdyt?

Reply:
That seems sensible.